### PR TITLE
Add workflow for testing breaking changes on CRDs (`Scaled{Object|Job}`)

### PR DESCRIPTION
**Pull request by JorTurFer**
This workflow will be triggered on every PR and also on every commit on main. 
It will build current code and push it to the testing packages, tagging the image with commit SHA. 
After that, the workflow will deploy KEDA (to local kind cluster) using the latest released version (by helm) and then a testing manifest will be deployed. If these steps work, current code will be released (manifest and code) and after that, the test-manifest will be redeployed, in case of breaking changes in CRDs, the workflow should fail in this point.

For the moment, I haven't added any smoke test to the workflow, but we could add them.

Notes: to avoid duplications, I used [reusable workflows](https://docs.github.com/es/actions/learn-github-actions/reusing-workflows). This makes that the template has to be in main for working (github referencing is crazy...)

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated

Fixes #

### CODE REVIEW

File: .github/assets/test-manifest.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Namespace
```
Comment (zroubalik) [inline .github/assets/test-manifest.yml:1]: I'd change the name of this file to be more descriptive, for example `test-upgrade-manifest.yaml` or something similar.

**Review by zroubalik — COMMENTED** [review]
This is great! 🚀 

Just a few very mino nits. 

And maybe we can add this workflow to the nightly workflow as well?

**Comment by JorTurFer** [comment]
> This is great! 🚀
> 
> Just a few very mino nits.
> 
> And maybe we can add this workflow to the nightly workflow as well?

Sure, I have added it to nightly workflow

File: .github/workflows/upgrade-validation-template.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        - kubernetesVersion: v1.21
+          kindImage: kindest/node:v1.21.1@sha256:69860bda5563ac81e3c0057d654b5253219618a22ec3a346306239bba8cfa1a6
+        - kubernetesVersion: v1.20
+          kindImage: kindest/node:v1.20.7@sha256:cbeaf907fc78ac97ce7b625e4bf0de16e3ea725daf6b04f930bd14c67c671ff9
+        - kubernetesVersion: v1.19
+          kindImage: kindest/node:v1.19.11@sha256:07db187ae84b4b7de440a73886f008cf903fcf5764ba8106a9fd5243d6f32729
+        - kubernetesVersion: v1.18
+          kindImage: kindest/node:v1.18.19@sha256:7af1492e19b3192a79f606e43c35fb741e520d195f96399284515f077b3b622c
+        - kubernetesVersion: v1.17
+          kindImage: kindest/node:v1.17.17@sha256:66f1d0d91a88b8a001811e2f1054af60eef3b669a9a74f9b6db871f2f1eeed00
```
Comment (tomkerkhove) [inline .github/workflows/upgrade-validation-template.yml:46]:
```suggestion
      fail-fast: false
      matrix:
        kubernetesVersion: [v1.22, v1.21, v1.20, v1.19, v1.18, v1.17]
        include:
          # Images are defined on every Kind release
          # See https://github.com/kubernetes-sigs/kind/releases
        - kubernetesVersion: v1.22
          kindImage: kindest/node:v1.22.0@sha256:b8bda84bb3a190e6e028b1760d277454a72267a5454b57db34437c34a588d047
        - kubernetesVersion: v1.21
          kindImage: kindest/node:v1.21.1@sha256:69860bda5563ac81e3c0057d654b5253219618a22ec3a346306239bba8cfa1a6
        - kubernetesVersion: v1.20
          kindImage: kindest/node:v1.20.7@sha256:cbeaf907fc78ac97ce7b625e4bf0de16e3ea725daf6b04f930bd14c67c671ff9
        - kubernetesVersion: v1.19
          kindImage: kindest/node:v1.19.11@sha256:07db187ae84b4b7de440a73886f008cf903fcf5764ba8106a9fd5243d6f32729
        - kubernetesVersion: v1.18
          kindImage: kindest/node:v1.18.19@sha256:7af1492e19b3192a79f606e43c35fb741e520d195f96399284515f077b3b622c
        - kubernetesVersion: v1.17
          kindImage: kindest/node:v1.17.17@sha256:66f1d0d91a88b8a001811e2f1054af60eef3b669a9a74f9b6db871f2f1eeed00
```

Comment (JorTurFer) [reply .github/workflows/upgrade-validation-template.yml:46]: I don't get the change :/
Is not `kubernetesVersion` already defined inside `include` section?

Comment (JorTurFer) [reply .github/workflows/upgrade-validation-template.yml:46]: I got it
https://docs.github.com/es/actions/learn-github-actions/workflow-syntax-for-github-actions#example-including-additional-values-into-combinations

Comment (tomkerkhove) [reply .github/workflows/upgrade-validation-template.yml]: Yes, but typically include is only if you need to add additional information to your combinations.

(I have to align our chart workflow with this as well, coming up)

**Review by zroubalik — APPROVED** [review]
LGTM, great!

**Pull request merged**